src/renderers/shared/fiber/ReactFiberBeginWork.js

```diff
@@ -365,9 +366,12 @@ module.exports = function<T, P, I, TI, C>(
     workInProgress.firstEffect = null;
     workInProgress.lastEffect = null;
 
-    if (workInProgress.progressedPriority === priorityLevel) {
+    if (workInProgress.progressedPriority === priorityLevel ||
+        priorityLevel === TaskPriority) {
```

Comment:
> Would always force unmounting children in an error boundary help with this case?

It would because we'd be recreating the whole tree instead of trying to reuse failed work.